### PR TITLE
Simplify ConsistencyLevel and CosmosStruct

### PR DESCRIPTION
**Pull request by rylev**
@thomastaylor312 and I are looking into simplifying the `cosmos` crate. This PR contains changes to two types that are indicative of the larger changes we would like to make. 

In general these changes favor concrete types over traits. Some additional changes I'd like to make include but are not limited to:
* Remove the `CosmosClient` trait and and rename `CosmosStruct` to `CosmosClient`. 
* Change the `hyper_client` field of `CosmosStruct` to `client` and make it a `Box<dyn Client>` . This is somewhat dependent on other work to abstract the client usage so implementations are not dependent on a particular http implementation.
* Consolidate all the clients into one client `CosmosClient` (aka `CosmosStruct`)

### CODE REVIEW

File: sdk/cosmos/src/clients/cosmos_struct.rs [hunk 1]
```diff
@@ -5,200 +5,108 @@ use crate::{
...
-            account: account,
-            auth_token,
-            cosmos_uri_builder,
-        })
+    pub fn new_emulator(address: &str, port: u16) -> Self {
+        //Account name: localhost:<port>
+        //Account key: C2y6yDjf5/R+ob0N8A7Cgv30VRDJIWEHLM+4QDU5DE2nQ9nDuVTqobD4b8mGGyPMbIZnqyMsEcaGQy67XIw/Jw==
+        let auth_token = AuthorizationToken::new_master(
+            "C2y6yDjf5/R+ob0N8A7Cgv30VRDJIWEHLM+4QDU5DE2nQ9nDuVTqobD4b8mGGyPMbIZnqyMsEcaGQy67XIw/Jw==",
+        ).unwrap();
```
Comment (guywaldman) [inline sdk/cosmos/src/clients/cosmos_struct.rs:55]: Remove?

File: sdk/cosmos/src/clients/cosmos_struct.rs [hunk 2]
```diff
@@ -428,6 +302,32 @@ fn generate_resource_link(u: &str) -> &str {
...
+    Public(String),
+    /// Azure China cloud
+    China(String),
+    // TODO: Other govt clouds?
+    /// A custom base URL
+    Custom { account: String, uri: String },
+}
+
+impl CloudLocation {
+    /// Consumes the location, returning a base URL
```
Comment (guywaldman) [inline sdk/cosmos/src/clients/cosmos_struct.rs:321]: Minor - but doesn't consume I believe

File: sdk/cosmos/src/clients/cosmos_struct.rs (hunk 1 above)
Comment (rylev) [reply sdk/cosmos/src/clients/cosmos_struct.rs:55]: Remove the unwrap? I don't believe we can. Can you explain?

Comment (guywaldman) [reply sdk/cosmos/src/clients/cosmos_struct.rs:55]: Never mind, just realized that this was a magic string for the Cosmos DB emulator account key, I thought this was a leftover from debugging. Ignore my silliness 🙃

Small suggestion, feel free to ignore if out of scope for the PR - document and refer to [the docs](https://docs.microsoft.com/en-us/azure/cosmos-db/local-emulator) for the emulator. Maybe move that connection string to a const. Might avoid some confusion for the uninitiated like myself.

**Comment by MindFlavor** [comment]
This is great work ❤️ . I think we probably want to merge [https://github.com/Azure/azure-sdk-for-rust/pull/79](https://github.com/Azure/azure-sdk-for-rust/pull/79) first since many modifications are probably going to create conflicts.

**Review by thomastaylor312 — APPROVED** [review]

**Pull request merged**